rs/ic_os/config/src/types.rs

```diff
@@ -61,6 +61,8 @@ pub struct GuestOSConfig {
 
 #[derive(Serialize, Deserialize, Debug, PartialEq, Eq, Clone)]
 pub struct ICOSSettings {
+    /// The node reward type determines node rewards
+    pub node_reward_type: String,
```

Comment:
node_reward_type exists as an enum in the registry protobuf: https://github.com/dfinity/ic/blob/4e46b92fc990ce07e6aa9e9706ea620c02a4be78/rs/protobuf/def/registry/node/v1/node.proto#L26

Rather than maintain the enum in two locations, we can leave it as a string here and just do a basic formatting check in the CreateSetuposConfig command

Reply:
This may be a slippery slope, but why can't we rely on these types directly?

We will need to do this mapping at some point when nodes use this config value to communicate with the registry - why not now? If the enum is in flux, it might be another good story to think about for config compatibility.

Reply:
Okay I actually looked at the enum this time 🤦‍♀️.

We _will_ still need some parsing to map the "human" names to these enum values, and at that point it would be good to use the types directly.

Until that mapping exists, though, this basic check in SetupOS looks fine. We just need to be wary that every variant that makes it past parsing needs to eventually be handled. (The same applies to the check from https://github.com/dfinity/ic/pull/1787.)